input/vlct/testing_utils.py

```diff
@@ -0,0 +1,303 @@
+# This file defines some useful testing utilities shared between a couple of
+# regression testing scripts
+
+import os
+import os.path
+import subprocess
+
+try:
+  basestring
+except NameError:
+  basestring = str
+
+import numpy as np
+
+def prep_cur_dir(executable):
+    cwd = os.getcwd()
+    if cwd[-10:] == "input/vlct":
+        os.chdir("../../")
+        # the following is just for a possible error message
+        orig_exec_path = os.path.join(cwd,"../../",executable)
+    else:
+        # the following is just for a possible error message
+        orig_exec_path = os.path.join(cwd,executable)
+
+    if not os.path.isfile(executable):
+        raise RuntimeError("Can't locate the executable: " + orig_exec_path)
+
+class _BaseCalcL1Norm(object):
```

Comment:
Yeah, the testing tools could definitely use some cleaning up. When I first made the issued the pull request, the future of the test framework wasn't really clear (at the meeting in May 2019 there was discussions about improving the framework to be more similar to Enzo, but not much happened until this past summer).

When I first started writing the tests, James has a bunch of tests executed by `SCons` that called command line utilities, checked the final output time of simulations, or check to see if the output files existed. I tried to follow this loose convention and wrote a command-line utility that computes L1 error norms, which can be found in `tools/l1_error_norms.py`. That way my tests could be easily adapted to the ultimate test framework.

The file you commented on has a lot of code because it's is wrapping `l1_error_norms.py`'s command line interface. There's different classes to support the 2 modes that `l1_error_norms.py` supports:
1. One mode computes takes the residuals from 2 snapshots and computes the L1 Error Norm from the result (I mostly use this for the linear wave test where I compare the inital conditions to the state after the waves have evolved over a full period)
2. The other mode compares a snapshot to a 1D table of values. (I use this for computing L1 error norms from shock tube test problems)

I think it would make a lot more sense to directly import functions defined in `tools/l1_error_norms.py`, directly. But, I'm not exactly a rush to do that before integrating the other parts #53 get's merged in. There's an additional challenge that also needs to be addressed, before we make this change. The command-line utility requires python 3 (way back I had found that there was a bug in the yt's enzo-e frontend that only appeared in python 2), while the version of `SCons` shipped with enzo-e requires python 2.7.

But, once that's all taken care of, I would be happy to clean this up.